src/schemas/hs00/WriterUntyped.cpp

```diff
@@ -10,51 +10,55 @@ namespace hs00 {
 WriterUntyped::ptr WriterUntyped::createFromJson(json const &Json) {
   if (Json.at("data_type") == "uint32") {
     return WriterUntyped::createFromJsonL1<uint32_t>(Json);
-  } else if (Json.at("data_type") == "uint64") {
+  }
+  if (Json.at("data_type") == "uint64") {
     return WriterUntyped::createFromJsonL1<uint64_t>(Json);
-  } else if (Json.at("data_type") == "double") {
+  }
+  if (Json.at("data_type") == "double") {
     return WriterUntyped::createFromJsonL1<double>(Json);
-  } else if (Json.at("data_type") == "float") {
+  }
+  if (Json.at("data_type") == "float") {
     return WriterUntyped::createFromJsonL1<float>(Json);
-  } else {
-    throw std::runtime_error(
-        fmt::format("unimplemented data_type: {}",
-                    Json.at("data_type").get<std::string>()));
   }
+  throw std::runtime_error(fmt::format(
+      "unimplemented data_type: {}", Json.at("data_type").get<std::string>()));
 }
 
 template <typename DataType>
 WriterUntyped::ptr WriterUntyped::createFromJsonL1(json const &Json) {
   if (Json.at("edge_type") == "uint32") {
     return WriterUntyped::createFromJsonL2<DataType, uint32_t>(Json);
-  } else if (Json.at("edge_type") == "uint64") {
+  }
+  if (Json.at("edge_type") == "uint64") {
```

Comment:
Did clang-tidy suggest replacing the else-ifs?

Reply:
Yes.

Reply:
I don't think I am very keen on that.
I think what this guy says makes sense:
https://youtu.be/otAcmD6XEEE?t=1724

tl;dw: unfocus your eyes and it looks like a bunch of guard statements followed by an action rather than flow control.